src/main/java/kr/hhplus/be/server/application/coupon/CouponOutboxProcessor.java

```diff
@@ -0,0 +1,116 @@
+package kr.hhplus.be.server.application.coupon;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import kr.hhplus.be.server.interfaces.web.coupon.dto.CouponIssuedEventDto;
+import kr.hhplus.be.server.domain.coupon.Coupon;
+import kr.hhplus.be.server.domain.coupon.CouponRepository;
+import kr.hhplus.be.server.infrastructure.external.coupon.CouponOutBoxEvent;
+import kr.hhplus.be.server.infrastructure.external.coupon.CouponOutBoxEventRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class CouponOutboxProcessor {
+
+    private final CouponOutBoxEventRepository outBoxEventRepository;
+    private final CouponRepository couponRepository;
+    private final ObjectMapper objectMapper;
+
+    private static final String COUPON_ISSUED_EVENT_TYPE = "COUPON_ISSUED";
+
+    /**
+     * 쿠폰 발급 이벤트 비동기 처리 (RDB 재고 동기화)
+     */
+    @Scheduled(fixedDelay = 10000) // 10초마다 실행
+    @Transactional
+    public void processCouponIssuedEvents() {
+        log.debug("쿠폰 발급 이벤트 비동기 처리 시작");
+        
+        try {
+            List<CouponOutBoxEvent> pendingEvents = outBoxEventRepository.findPendingEventsByType(COUPON_ISSUED_EVENT_TYPE);
```

Comment:
```suggestion
            List<CouponOutBoxEvent> pendingEvents = outBoxEventRepository.findPendingEventsByType(COUPON_ISSUED_EVENT_TYPE, 100);
```

배치 크기를 제한 두는것도 좋을것 같습니다

추가적으로 조회는 트랜잭션 범위 밖에 있어도 될것같아요!